File: Graph/498.diagonal-traverse.go
package leetcode

// @leet start

func findDiagonalOrder(mat [][]int) []int {
	nrow, ncol := len(mat), len(mat[0])
	nums := make([][]int, nrow+ncol-1)
	for r := range nrow {
		for c := range ncol {
			nums[r+c] = append(nums[r+c], mat[r][c])
		}
	}
	res := make([]int, 0, nrow*ncol)
	for i, num := range nums {
		if i%2 == 0 {
		}
		res = append(res, num...)
	}
	return res
}

// @leet end

File: Graph/498.diagonal-traverse.py
from typing import List


# @leet start
class Solution:
    def findDiagonalOrder(self, mat: List[List[int]]) -> List[int]:
        nr = len(mat)
        nc = len(mat[0])
        diagL = [[] for _ in range(nr + nc - 1)]

        for r in range(nr):
            for c in range(nc):
                diagL[r + c].append(mat[r][c])
        res = []
        for i, row in enumerate(diagL):
            if i % 2 == 0:
                row.reverse()
            res.extend(row)
        return res


# @leet end
